File: All-Topics/1-50/14-LongestCommonPrefix.java
class Solution {
  public String longestCommonPrefix(String[] strs) {

    int length = strs[0].length();
    int count = 0;
    String result = "";
    int min = Integer.MAX_VALUE;
    for (int i = 0; i < strs.length; i++) {
      if (strs[i].length() < min) {
        min = strs[i].length();
      }
    }

    while (count < min) {
      char element = strs[0].charAt(count);
      for (int i = 0; i < strs.length; i++) {
        if (element != strs[i].charAt(count)) {
          return result;
        }
      }
      result += element;
      count++;
    }
    return result;
  }
}

File: All-Topics/1-50/14-LongestCommonPrefix.cpp
#include <vector>
#include <string>
#include <limits>



class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        std::string result="";
        int size=1;
        int minLength=std::numeric_limits<int>::max();

        for(std::string val:strs){
            if(val.size()<minLength)minLength=val.size();
        }


        for(int i=0;i<minLength;i++){
            std::string part=strs[0].substr(0, size);
            for(std::string val:strs){
                std::string part2=val.substr(0, size);
                if(part!=part2){
                    return result;
                }
            }
            result+=strs[0][i];
            size++;
        }


        return result;







    }
};